src/NLog.Targets.Syslog/MessageCreation/MessageBuilder.cs

```diff
@@ -62,7 +72,15 @@ public void PrepareMessage(ByteArray buffer, LogEventInfo logEvent, string logEn
         private static string Pri(Facility facility, Severity severity)
         {
             var priVal = (int)facility * 8 + (int)severity;
-            return $"<{priVal.ToString(CultureInfo.InvariantCulture)}>";
-        }
+            if (PriValueCache.TryGetValue(priVal, out var priOutput))
+                return priOutput;
+            
+            return GeneratePriOutput(priVal);
```

Comment:
In which case an output will not be in the cache?

Reply:
Maybe if taking a random integer and casting to `Facility`.

Reply:
You are right but I think the risk can be taken